File: learn_py_with_js/email/email.mjs
const emailAddresses = [
    "john.doe@example1.com",
    "jane.smith@example2.com",
    "foo@bar.com",
];

const regex = /([^@]+)@(.+)/

emailAddresses.forEach(email => {
    const [_,username, domian] = email.match(regex)
    console.log(`username: ${username}, domian: ${domian}`)
})

File: learn_py_with_js/email/email.py
import re
emailAddresses = [
    "john.doe@example1.com",
    "jane.smith@example2.com",
    "foo@bar.com",
];
regex = re.compile(r'([^@]+)@(.+)')

for email in emailAddresses:
    match = regex.match(email)
    if match:
        username, domain = match.groups()
        print(f"Username: {username}, Domain: {domain}")